### PR TITLE
[Fix #17273][V4] Throw error when Tether is not included

### LINKED ISSUE #17273: Tooltips: "Tether is not defined"
Today I wanted to try the Bootstrap 4 alpha and when I want to use the tooltips plugin I get this error in my debugger console:

```
Uncaught ReferenceError: Tether is not defined
bootstrap.js:2967

show @ bootstrap.js:2967
_enter @ bootstrap.js:316
4n.extend.proxy.n.isFunction.e @ jquery-1.11.0.min.js:2
n.each.n.event.special.(anonymous function).handle @ jquery-1.11.0.min.js:3
n.event.dispatch @ jquery-1.11.0.min.js:3
n.event.add.r.handle @ jquery-1.11.0.min.js:3
```

### PR DESCRIPTION
**Pull request by Johann-S**
Hi, 

it's a fix for #17273

### CODE REVIEW

**Comment by cvrebert** [comment]
* There isn't guaranteed to be a `window.Tether` if UMD is in use. (For that matter, we probably ought to be explicitly `import`ing Tether.)
* Seems like we could do the check just once, before defining the Tooltip class, rather than re-checking every time a new tooltip is instantiated.

**Comment by Johann-S** [comment]
I thought about adding Tether with this code : 
``})(jQuery, Tether)`` at the end of Tooltip src but we cannot customized the error message

To import Tether is it possible to do it like how you import ``utils.js`` ?

**Comment by Johann-S** [comment]
Maybe this check can be done with same way to check if JQuery is included or not 
https://github.com/twbs/bootstrap/blob/v4-dev/Gruntfile.js#L44-L46

**Comment by fat** [comment]
i went ahead and just merged a simple check at the start of the script.

it checks window which isn't great if peopel are using UMD - but also we aren't properly importing 3rd party deps yet with our code, so i figure that can be cleaned up at that point

**Comment by Johann-S** [comment]
Maybe it would be useful to add my unit test to check the import of Tether

File: js/src/tooltip.js [hunk 1]
```diff
@@ -415,6 +415,9 @@ const Tooltip = (($) => {
     }
 
     _setListeners() {
+      if (window.Tether === undefined) {
+        throw new Error('Bootstrap tooltips require Tether http://github.hubspot.com/tether/')
```
Comment (tracker1) [inline js/src/tooltip.js]:
```
if (typeof Tether === undefined) {
```

I'm using webpack provider to provide `window.Tether`, and `Tether` would prefer the latter though.

**Comment by bardiharborow** [comment]
Superseded by d600831.

**Pull request closed without merging**